pkg/operator/ceohelpers/bootstrap_test.go

```diff
@@ -0,0 +1,362 @@
+			operatorState: newOperatorState().
+				WithLatestRevision(1).
+				WithNodeAtCurrentRevision("node-1", 1).
+				WithNodeAtCurrentRevision("node-2", 1).
+				Build(),
+			expectError: nil,
+		},
+		"non-HA with insufficient nodes during bootstrap": {
+			namespace:          newNamespace(operatorclient.TargetNamespace).WithNonHAAnnotation().Build(),
+			bootstrapConfigMap: newBootstrapConfigMap().WithProgressingStatus().Build(),
+			operatorState: newOperatorState().
+				WithLatestRevision(1).
+				WithNodeAtCurrentRevision("node-1", 1).
+				Build(),
+			expectError: fmt.Errorf("not enough nodes"),
+		},
+		"non-HA with sufficient nodes during steady state": {
+			namespace:          newNamespace(operatorclient.TargetNamespace).WithNonHAAnnotation().Build(),
+			bootstrapConfigMap: newBootstrapConfigMap().WithCompleteStatus().Build(),
+			operatorState: newOperatorState().
+				WithLatestRevision(1).
+				WithNodeAtCurrentRevision("node-1", 1).
+				WithNodeAtCurrentRevision("node-2", 1).
+				WithNodeAtCurrentRevision("node-3", 1).
+				Build(),
+			expectError: nil,
+		},
+		"non-HA with insufficient nodes during steady state": {
+			namespace:          newNamespace(operatorclient.TargetNamespace).WithNonHAAnnotation().Build(),
+			bootstrapConfigMap: newBootstrapConfigMap().WithCompleteStatus().Build(),
+			operatorState: newOperatorState().
+				WithLatestRevision(1).
+				WithNodeAtCurrentRevision("node-1", 1).
+				WithNodeAtCurrentRevision("node-2", 1).
+				Build(),
+			expectError: fmt.Errorf("not enough nodes"),
+		},
+	}
+
+	for name, test := range tests {
+		t.Run(name, func(t *testing.T) {
+			namespaceIndexer := cache.NewIndexer(cache.MetaNamespaceKeyFunc, cache.Indexers{cache.NamespaceIndex: cache.MetaNamespaceIndexFunc})
+			if test.namespace != nil {
+				if err := namespaceIndexer.Add(test.namespace); err != nil {
+					t.Fatal(err)
+				}
+			}
+			configmapIndexer := cache.NewIndexer(cache.MetaNamespaceKeyFunc, cache.Indexers{cache.NamespaceIndex: cache.MetaNamespaceIndexFunc})
+			if test.bootstrapConfigMap != nil {
+				if err := configmapIndexer.Add(test.bootstrapConfigMap); err != nil {
+					t.Fatal(err)
+				}
+			}
+			fakeNamespaceMapLister := corev1listers.NewNamespaceLister(namespaceIndexer)
+			fakeConfigMapLister := corev1listers.NewConfigMapLister(configmapIndexer)
+			fakeStaticPodClient := v1helpers.NewFakeStaticPodOperatorClient(test.operatorState.Spec, test.operatorState.Status, nil, nil)
+			actualErr := EnsureSafeToScaleCluster(fakeConfigMapLister, fakeStaticPodClient, fakeNamespaceMapLister)
+			if test.expectError != nil && actualErr == nil {
+				t.Errorf("expected error=%v, got %v", test.expectError, actualErr)
+			}
+			if test.expectError == nil && actualErr != nil {
+				t.Errorf("expected error=%v, got %v", test.expectError, actualErr)
+			}
+		})
+	}
+}
+
+type bootstrapConfigMapBuilder struct {
+	corev1.ConfigMap
+}
+
+func newBootstrapConfigMap() bootstrapConfigMapBuilder {
```

Comment:
(No action required) This is a neat pattern for more complicated use cases, but maybe for the namespace and configmap?

```golang
func newBoostrapConfigMap(status string) *corev1.ConfigMap {
 ...
}
```

Reply:
Upon further consideration based on your other comments about the fixtures, I decided the builders are overkill in general and removed all of them